### PR TITLE
Docs collate for GitHub Actions

### PR DESCRIPTION
**Pull request by zdmytriv**
## what
* Docs collate for GitHub Actions

## why
* Docs collate for GitHub Actions

## references
*

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # Usage for docs-collator
 
-Document collation supports 2 types of projects:
+Document collation supports 3 types of projects:
 - components - `cloudposse/terraform-aws-component` - components from single repo `cloudposse/terraform-aws-component` under `modules/` dir 
 - modules - `cloudposse/terraform-*` - for all public (non archived) repos except `cloudposse/terraform-aws-component` that are prefixed with `terraform-`
+- github actions - `cloudposse/github-action-*` - for all public (non archived) repos that are prefixed with `github-action-`
```
Comment (osterman) [inline scripts/README.md:6]:
```suggestion
- github actions - `cloudposse/github-action-*` - for all public (non-archived) repos that are prefixed with `github-action-`
```

File: scripts/docs-collator/render_docs_for_modules.py [hunk 2]
```diff
@@ -47,9 +46,9 @@ def main(github_api_token, output_dir, download_dir, includes_csv, excludes_csv,
 @click.option('--github-api-token',
               envvar='PUBLIC_REPO_ACCESS_TOKEN', required=True, help="Github API token")
 @click.option('--output-dir',
-              default=OUTPUT_DOC_DIR, required=True, help="Rendered component output dir")
+              default=OUTPUT_DOC_DIR, required=True, help="Rendered doc output dir")
 @click.option('--download-dir',
-              default=DOWNLOAD_TMP_DIR, required=True, help="Temporary output dir")
+              default=DOWNLOAD_TMP_DIR, required=True, help="Temporary download dir")
```
Comment (osterman) [inline scripts/docs-collator/render_docs_for_modules.py:51]:
```suggestion
              default=DOWNLOAD_TMP_DIR, required=True, help="Temporary download directory")
```

File: scripts/docs-collator/render_docs_for_modules.py [hunk 3]
```diff
@@ -47,9 +46,9 @@ def main(github_api_token, output_dir, download_dir, includes_csv, excludes_csv,
 @click.option('--github-api-token',
               envvar='PUBLIC_REPO_ACCESS_TOKEN', required=True, help="Github API token")
 @click.option('--output-dir',
-              default=OUTPUT_DOC_DIR, required=True, help="Rendered component output dir")
+              default=OUTPUT_DOC_DIR, required=True, help="Rendered doc output dir")
```
Comment (osterman) [inline scripts/docs-collator/render_docs_for_modules.py:49]:
```suggestion
              default=OUTPUT_DOC_DIR, required=True, help="Rendered documentation output directory")
```

File: scripts/docs-collator/render_docs_for_modules.py [hunk 4]
```diff
@@ -65,7 +64,7 @@ def cli_main(github_api_token, output_dir, download_dir, includes, excludes, fai
                         datefmt='%d-%m-%Y %H:%M:%S',
                         level=logging.getLevelName(log_level))
 
-    logging.info(f"Tmp dir: {download_dir}, components dir: {output_dir}")
+    logging.info(f"Tmp dir: {download_dir}, doc output dir: {output_dir}")
```
Comment (osterman) [inline scripts/docs-collator/render_docs_for_modules.py:67]:
```suggestion
    logging.info(f"Tmp dir: {download_dir}, documentation output directory: {output_dir}")
```

File: scripts/render-docs-for-github-actions.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -e
+
+usage() {
+	echo "Required env var '$1' is missing"
```
Comment (osterman) [inline scripts/render-docs-for-github-actions.sh:6]:
```suggestion
	echo "Required environment variable '$1' is missing"
```

File: scripts/docs-collator/templates/github-actions/readme.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+## Deprecated
+
+{{ if has (ds "config").deprecated "notice" }}
+  {{- (ds "config").deprecated.notice }}
+{{- else }}
+  This module is no longer actively maintained
+{{- end }}
+
+{{ if has (ds "config") "description" }}
+### Historical Description
```
Comment (osterman) [inline scripts/docs-collator/templates/github-actions/readme.md:33]: Why is it the "Historical" description?

File: scripts/docs-collator/render_docs_for_github_actions.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            else:
+                logging.warning(e.message)
+                continue
+
+
+@click.command()
+@click.option('--github-api-token',
+              envvar='PUBLIC_REPO_ACCESS_TOKEN', required=True, help="Github API token")
+@click.option('--output-dir',
+              default=OUTPUT_DOC_DIR, required=True, help="Rendered doc output dir")
```
Comment (osterman) [inline scripts/docs-collator/render_docs_for_github_actions.py:50]:
```suggestion
              default=OUTPUT_DOC_DIR, required=True, help="Rendered documentation output directory")
```

File: scripts/docs-collator/render_docs_for_github_actions.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+                logging.error(e.message)
+                sys.exit(1)
+            else:
+                logging.warning(e.message)
+                continue
+
+
+@click.command()
+@click.option('--github-api-token',
+              envvar='PUBLIC_REPO_ACCESS_TOKEN', required=True, help="Github API token")
```
Comment (osterman) [inline scripts/docs-collator/render_docs_for_github_actions.py:48]:
```suggestion
              envvar='PUBLIC_REPO_ACCESS_TOKEN', required=True, help="GitHub API token")
```

File: .github/workflows/website-deploy-preview.yml [hunk 9]
```diff
@@ -77,6 +77,12 @@ jobs:
         run: |
           ./scripts/render-docs-for-modules.sh
 
+      - name: "Render docs for github actions"
```
Comment (osterman) [inline .github/workflows/website-deploy-preview.yml:80]:
```suggestion
      - name: "Render Documentation for GitHub Actions"
```

File: .github/workflows/website-deploy-release.yml [hunk 10]
```diff
@@ -74,6 +74,12 @@ jobs:
         run: |
           ./scripts/render-docs-for-modules.sh
 
+      - name: "Render docs for github actions"
```
Comment (osterman) [inline .github/workflows/website-deploy-release.yml:77]:
```suggestion
      - name: "Render Documentation for GitHub Actions"
```

File: content/github-actions/index.md [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+sidebar_label: Overview of GitHub Actions
+description: GitHub Actions
+sidebar_position: 0
+---
+
+This is a collection of reusable GitHub Actions.
+
+## Introduction
+
+In this catalog you'll find real-world examples of how we've implemented various common github actions.
```
Comment (osterman) [inline content/github-actions/index.md:12]:
```suggestion
In this library you'll find real-world examples of how we've implemented reusable GitHub Actions to solve common CI/CD challenges.
```

File: scripts/docs-collator/templates/github-actions/readme.md (hunk 6 above)
Comment (zdmytriv) [reply scripts/docs-collator/templates/github-actions/readme.md:33]: I took the template from here https://github.com/cloudposse/build-harness/blob/master/templates/README.md.gotmpl#L60 and simplified it.

**Review by osterman — APPROVED** [review]

**Pull request merged**